woocommerce/woocommerce
# Migrate simple product template to HTML

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This adds some code to convert existing templates to comment-delimited format. The code is not used in production, but is being used manually to support the migration of the existing templates. 

Its result is in the `simple.php` file. The file was also manually formatted.

Part of #47999 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the new Product Editor
2. Enable the `product-editor-template-system` feature flag
  a. Install the WooCommerce Beta Tester plugin
  b. Go to the Settings -> WCA Test Helper -> Features
  c. Enable the `product-editor-template-system` feature flag

![image](https://github.com/woocommerce/woocommerce/assets/77539/4fbfeeae-3ed2-417f-8419-ef813d3a0e0a)

3. Update the Simple product_form post
  a. Install and activate the plugin [pft-dev-tools](https://github.com/woocommerce/pft-dev-tools/archive/refs/heads/trunk.zip) (note that this plugin forces the `product-editor-template-system` feature flag to be enabled)
  b. Generate the `Simple` form template by running the following `wp-cli` command:

```sh
wp trigger upgrader_process_complete --action=install
```

4. Create a new product from the new Product Editor
5. Click on the `Change product type` link/button
6. Confirm you see the Simple template there
7. Select it
8. Check that the template is loaded with all the fields



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
